prow/apis/prowjobs/v1/types.go

```diff
@@ -163,13 +163,18 @@ type ProwJobSpec struct {
 	ReporterConfig *ReporterConfig `json:"reporter_config,omitempty"`
 
 	// RerunPermissions holds information about which users can rerun the job
-	RerunPermissions *RerunPermissions `json:"rerun_permissions,omitempty"`
+	RerunPermissions RerunPermissions `json:"rerun_permissions,omitempty"`
```

Comment:
To safely make this change we may need to have a migration period where any `rerun_permissions: nil` fields are converted to `rerun_permissions: {}`, otherwise parsing will fail when we try to parse `nil` into a `RerunPermissions` struct.
@stevekuznetsov

Reply:
Never mind, I think the `omitempty` saves us. We shouldn't have serialized to `rerun_permissions: nil` anywhere.